plugins/module_utils/mso.py

```diff
@@ -879,6 +887,8 @@ def dict_from_ref(self, data):
                     'contracts': ['contractName', 'schemaId', 'templateName'],
                     'l3outs': ['l3outName', 'schemaId', 'templateName'],
                     'anps': ['anpName', 'schemaId', 'templateName'],
+                    'serviceGraphs':['serviceGraphName', 'schemaId', 'templateName'],
+                    'serviceNodes':['serviceNodeNmae','serviceGraphName', 'schemaId', 'templateName'],
```

Comment:
If it is 4 things the recursive part should take care of it. Also there is a typo in the first portion.

Reply:
done